lib/checkother.cpp

```diff
@@ -1391,6 +1391,59 @@ void CheckOther::checkConstVariable()
         // Skip if address is taken
         if (Token::findmatch(var->nameToken(), "& %varid%", scope->bodyEnd, var->declarationId()))
             continue;
+        // Skip if another non-const variable is initialized with this variable
+        {
+            //Is it the right side of an initialization of a non-const reference
+            bool usedInAssignment = [&]
+                {
```

Comment:
Imho, this outer lambda should be removed.

Reply:
Originally I didn't have it but I ended up wanting to ```continue``` the outer for loop from inside the new inner one which I can't do. However I liked the pattern of breaking out as soon as you know you've got something so I used return instead. That meant I needed something to return from.

I could instead have started with ```bool usedInAssignment = false;``` and then set it to true in my loop and added an extra condition on usedInAssignment to the loop in order to break out early. Personally I would find that code harder to read as it's less clear that the whole point of that block of code is to calculate this one boolean and it would mean the boolean would exist in an intermediate state which I also don't like.

Let me know your thoughts on this and if you insist then I'll change it.

Reply:
actually since I can just copy-paste your solution from below I've done that